clean-code-challanges/src/main/examples/PigLatinTranslator.java

```diff
@@ -18,6 +18,40 @@
 public class PigLatinTranslator {
 
   public String translate(String englishPhrase) {
-    return null;
+    String result = "";
+    String attachment = "ay";
+    int position = 0;
+    String separator = "\\s+";
+    String [] words = englishPhrase.split(separator);
+
+    if (englishPhrase.contains(" ")) {
+      attachment = "ay ";
+    }
+
+    for (int i = 0; i < words.length; i++) {
+      for (char j = 0; j < words[i].length(); j++) {
+        if (isVowel(words[i].charAt(j))) {
+          position = j;
+          break;
+        }
+      }
+      if (isVowel(words[i].charAt(0)) || words[i].substring(0, 2).equals("xr") || words[i].substring(0, 2).equals("yt")) {
+        result += words[i] + attachment;
+      } else {
+        if (words[i].substring(1, 3).equals("qu")) {
+          result += words[i].substring(3) + words[i].substring(0, 3) + attachment;
+        } else {
+          result += words[i].substring(position) + words[i].substring(0, position) + attachment;
```

Comment:
![image](https://user-images.githubusercontent.com/54735479/93789546-47b79780-fc32-11ea-886c-0544a156c86e.png)

Reply:
almost the right picture :P
![image](https://user-images.githubusercontent.com/54737443/94287350-76957c80-ff56-11ea-8139-8f3df5a2bde6.png)
 
Ich müsste allgemein die Bedingungen in Methoden auslagern.